### PR TITLE
Rice compression support

### LINKED ISSUE #9: Request: Rice compression support
Related: https://www.astromatic.net/forum/showthread.php?tid=597

At least one pipeline the Euclid SGS relies on a version of SExtractor with support for Rice compression. This version is built from the [Roy branch](https://github.com/astromatic/sextractor/tree/RoyBranch) if I'm not mistaken. It would be good if the current processing environment of Euclid could just use the latest and the greatest version of SExtractor (which is a bit more accurate in my testing). Hence this feature request for Rice compression support.

I managed rebase the Roy Branch on top of current master branch, and I'd be happy to put in a PR for this. But I'm curious why this has not been merged yet into master. It is admittedly rough around the edges (e.g. non-optional CFITSIO dependency and a hackish compression detection) and I can try my best to polish these. But are there any other blocking issues with this branch?

Note: from my understanding it is a no-no for the respective Euclid pipeline to first unpack with e.g. `funpack` and the process with SExtractor, so build-in Rice compression is essential for them.

### PR DESCRIPTION
**Pull request by teake**
Closes #9.

This is work continued from the [RoyBranch](https://github.com/astromatic/sextractor/tree/RoyBranch) branch, and adds optional CFITSIO support. It adds the following `configure` switches:

- `--enable-cfitsio`
- `--with-cfitsio-libdir`
- `--with-cfitsio-incdir`

Also, the CFITSIO dependent code is conditionally included in `#ifdef HAVE_CFITSIO` statements, and I have refactored to bundle those `#ifdef`s together as much as I could. Ideally, I'd like to see  all `#ifdef HAVE_CFITSIO` statements in one source file (e.g. `fitsio_support.c`), but I couldn't find a good way to do this.

Apart from the `#ifdef`s the C code has only been refactored, and has no functional differences from the RoyBranch.

The build-in test passes, and I have run a couple of other tests as well (both with and without CFITSIO support).

Note: the changes to `configure` from running `autogen.sh` are not included in this PR.

### CODE REVIEW

File: src/fits/fitsbody.c [hunk 1]
```diff
@@ -344,7 +443,8 @@ void	read_body(tabstruct *tab, PIXTYPE *ptr, size_t size)
         if (spoonful>size)
           spoonful = size;
         bufdata = (char *)bufdata0;
-        QFREAD(bufdata, spoonful*tab->bytepix, cat->file, cat->filename);
+        readSpoonful(tab, spoonful, bufdata0, bufdata);
```
Comment (teake) [inline src/fits/fitsbody.c]: Here `bufdata0` is of type `int *` but the function expects `'double *`. Not sure if that's a problem.

**Comment by ebertin** [comment]
Thank you @teake, and sorry for the delay. I wanted to review the code first. What do you think of the BZERO and BSCALE used as doubles? Since the result will always be PIXTYPE, I don't see the benefit of using something else than PIXTYPEs (which are floats by default). Doubles may have a noticeable impact on speed (after auto-vectorization). Thanks.

**Comment by teake** [comment]
Just to be clear: you're talking about `structtab::bscale` and `structtab::bzero`, not the ones in `readTileCompressed`?

In that case, I'm not sure. In the current master branch I would agree, but in 3f06d390ca74be4c979e80f0d33c5d1449adb4ff rhenders removed the cast to `PIXTYPE` in `read_body` with the comment _"this cast from double to float loses precision"_. I don't know what the reason was -- maybe (s)he was seeing some actual loss in precision or just thought it would happen never tested it. So without contacting rhenders (not [github.com/rhenders](https://github.com/rhenders) I presume) or testing this out ourselves I don't feel confident to make a statement about this.

But if you do feel confident about it or can argue it's not a problem, I don't object :).

**Comment by ebertin** [comment]
Yes precisely that 3f06d39 change to `read_body()`. I think that the change to double may have been an attempt to mitigate rounding issues often seen with the quantization associated to the floating point compression algorithm, e.g., 0 becomes 1e-5 or something like that. Unfortunately even the `fcompress` utility has this issue, and I doubt moving to doubles will solve it.
Nevertheless I agree with you it is safer to leave it like this until further notice. I will check that it does not excessively impact speed.
By any chance, have you run some basic tests on this version yourself?

**Comment by teake** [comment]
Yes, apart from the built-in `make check` I ran a test reproducing an example sextractor call from the Euclid EXT pipeline and that passed. In addition, the sextractor unit tests from Astro-WISE also pass, although they are quite limited in scope.

**Comment by teake** [comment]
Btw, the test from the Euclid EXT pipeline uses Rice compression -- the others don't.

**Comment by teake** [comment]
@ebertin Just pinging you on the state of this MR. Do you need any feedback or changes from me or is this good to go?

**Pull request merged**